nautobot/docs/models/dcim/deviceredundancygroups.md

```diff
@@ -0,0 +1,250 @@
+### Retrieving the data - Secondary Failover Unit ("nyc-fw-secondary")
+
+An example data returned from Nautobot is presented below.
+
+```python
+>>> secondary_unit_details = get_gql_failover_details("nyc-fw-secondary").json
+```
+
+```json
+{'data': {'devices': [{'name': 'nyc-fw-secondary', 'device_redundancy_group': {'name': 'nyc-firewalls', 'members': [{'name': 'nyc-fw-primary', 'device_redundancy_group_priority': 100, 'interfaces': [{'type': 'VIRTUAL', 'name': 'failover-link', 'ip_addresses': [{'host': '172.27.48.0', 'prefix_length': 31}], 'parent_interface': {'name': 'gigabitethernet0/3', 'type': 'A_1000BASE_T'}}]}, {'name': 'nyc-fw-secondary', 'device_redundancy_group_priority': 50, 'interfaces': [{'type': 'VIRTUAL', 'name': 'failover-link', 'ip_addresses': [{'host': '172.27.48.1', 'prefix_length': 31}], 'parent_interface': {'name': 'gigabitethernet0/3', 'type': 'A_1000BASE_T'}}]}]}}]}}
+```
+
+### Processing the data - Primary Failover Unit ("nyc-fw-primary")
+
+Before rendering the data, we will process the data to simplify data variables within template:
+
+```python
+>>> hostname = primary_unit_details['data']['devices'][0]['name']
+>>> hostname
+'nyc-fw-primary'
+>>>
+>>> failover_members = primary_unit_details['data']['devices'][0]['device_redundancy_group']['members']
+>>>
+>>> device = [failover_member for failover_member in failover_members if failover_member['name'] == hostname][0]
+>>>
+>>> device
+{'name': 'nyc-fw-primary', 'device_redundancy_group_priority': 100, 'interfaces': [{'type': 'VIRTUAL', 'name': 'failover-link', 'ip_addresses': [{'address': '172.27.48.0/31'}], 'parent_interface': {'name': 'gigabitethernet0/3', 'type': 'A_1000BASE_T'}}]}
+>>>
+>>> failover_peer = [failover_member for failover_member in failover_members if failover_member['name'] != hostname][0]
+>>>
+>>> device_failover_interface = device['interfaces'][0]
+>>> peer_failover_interface = failover_peer['interfaces'][0]
+```
+
+### Creating Cisco ASA Configuration Template - Common for Primary and Secondary Units
+
+Following snippet represents an example Juniper BGP Configuration Template:
+
+```python
+# Configuration Template for Cisco ASA
+template_body="""
+!
+hostname {{ device.name }}
+!
+failover lan unit {{ priority_mapping.device_redundancy_group_priority }}
+failover lan interface {{ device.interfaces[0].name }} {{ device.interfaces[0].parent_interface.name }} 
+!
+failover interface ip {{ device.interfaces[0].name }} {{ device_failover_interface.ip_addresses[0].host }}/{{ device_failover_interface.ip_addresses[0].prefix_length }} standby {{ peer_failover_interface.ip_addresses[0].host }}
+interface {{ device_failover_interface.parent_interface.name }} 
+  no shutdown
+!
+failover link {{ device.interfaces[0].name }} {{ device_failover_interface.parent_interface.name }} 
+!
+!failover ipsec pre-shared-key !Nautobot Secrets
+!
+failover
+!
+"""
+```
+
+### Rendering Juniper Jinja2 BGP Configuration Template with the data retrieved from GraphQL
```

Comment:
Cisco ASA not Juniper BGP, presumably

Reply:
Thanks! Fixed - was having some git/Pycharm issues yesterday and committed un-completed version :-(